### ISSUE TITLE
Late media - SDP ACKed immediately

### ISSUE THREAD

**Issue by IdlePhysicist** [drachtio/drachtio-srf#104]
In my use case Drachtio is a B2BUA routing calls from an LCS to various systems.  I am now implementing the reverse, the primary difference is that these systems that are dialling out use "late media".

As you can see in the attached sip-ladder (see image below, pcap via email) the LCS (10.3.40.12) returns a 200 OK with SDP candidates but Drachtio (10.89.4.193) responds immediately with an ACK before forwarding the 200 OK to the originating system. Media timeout then occurs because no media has been appropriately selected (as far as the LCS is concerned).

![sip-ladder](https://user-images.githubusercontent.com/20644761/115040334-3b927080-9ec9-11eb-8446-840048e0fa76.png)

What should be occurring (in my mind) is the following:
```
A               Drachtio                B
|                   |<---200 OK (SDP)---|
|<---200 OK (SDP)---|                   |
|----ACK (SDP)----->|                   |
|                   |----ACK (SDP)----->|
```

<details><summary>Sample Code from client application</summary>
<p>

```js
app.sip.ep.invite((req, res) => {
    let dnis = req.calledNumber
    var route

    if (app.route.config[dnis]) {
      // logic to find the next hop `new_to` for the call.
    }
    var new_to = `sip:${dnis}@${route}`

    app.sip.ep.createB2BUA(req, res, new_to,
      {
        localSdpB: req.body,
        headers: {
          'X-Virtual-Edge': 'Private'
        },
        responseHeaders: {
          'X-Virtual-Edge': 'Private'
        },
        proxyRequestHeaders: ['Via', 'From', 'Contact', 'Allow', 'Supported', 'User-agent', 'Content-Type'],
        proxyResponseHeaders: ['Accept', 'Allow', 'Record-Route', 'Session-Expires']
      })
    .then(({uas: a_leg, uac: b_leg}) => {

      a_leg.on('destroy', () => { b_leg.destroy() })
      b_leg.on('destroy', () => { a_leg.destroy() })

      b_leg.on ('modify', (req, res) => {
        a_leg.modify(req.body)
        .then((sdp) => {
          res.send(200, {body: sdp})
        })
      })
    })
    .catch((err) => {
      res.send(408, `non-success`)
    })
  })
```

</p>
</details>

**Comment by davehorton** [comment]
could you install drachti@4.4.49-beta.5 from npmjs and test your scenario?  I've made a fix there that should address your scenario.

**Comment by IdlePhysicist** [comment]
Apologies from only getting back to you now. I have installed Drachtio-srf version 4.4.49-beta.5, I can confirm that I am still not seeing the correct behaviour. In this case the LCS (for reasons unknown to me) returned only one SDP candidate this time.

![Screenshot from 2021-04-26 16-56-55](https://user-images.githubusercontent.com/20644761/116117190-df430400-a6b3-11eb-80e7-d578cdf13497.png)

As you can see from the above image, Drachtio is still immediately ACK-ing the 200 OK (containing the SDP candidate(s)). Only this time it is an empty ACK which acknowledges the single SDP candidate.

**Comment by davehorton** [comment]
Can you run your scenario with DEBUG=drachtio:* and provide me with the application log output as well as the drachtio server log please.

**Comment by IdlePhysicist** [comment]
I have reviewed our testing, and the 4.4.49-beta.5 is working as expected and intended. Sorry for the confusion earlier, our test was broken not the srf.

![Screenshot from 2021-04-26 18-29-32](https://user-images.githubusercontent.com/20644761/116125418-5f219c00-a6bd-11eb-9735-5199873fd926.png)

**Comment by davehorton** [comment]
This is fixed in drachtio-srf@4.4.49
